felix/netlinkshim/mocknetlink/netlink.go

```diff
@@ -76,21 +78,55 @@ var _ netlinkshim.Interface = (*MockNetlinkDataplane)(nil)
 var (
 	SimulatedError        = errors.New("dummy error")
 	NotFoundError         = errors.New("not found")
+	LinkNotFoundError     = netlink.LinkNotFoundError{}
 	FileDoesNotExistError = errors.New("file does not exist")
 	AlreadyExistsError    = errors.New("already exists")
 	NotSupportedError     = errors.New("operation not supported")
 )
 
+func init() {
+	// Ugh, the error field isn't exported and logging out the error
+	// panics if the error field isn't set.  Use an unsafe cast to
+	// set the value.
+
+	// Copy of the netlink.LinkNotFoundError struct.
+	type myLinkNotFoundError struct {
+		error
+	}
+
+	// First check that our struct matches the netlink one...
+	nlType := reflect.TypeOf(LinkNotFoundError)
+	ourType := reflect.TypeOf(myLinkNotFoundError{})
+	if nlType.NumField() != ourType.NumField() {
+		panic("netlink.LinkNotFoundError structure appears to have changed (different number of fields)")
+	}
+	for i := 0; i < ourType.NumField(); i++ {
+		nlFieldType := nlType.Field(i).Type
+		outFieldType := ourType.Field(i).Type
```

Comment:
do you mean "our" instead of "out" ?

Reply:
Ah yes, that's one of my recurring typos :grimacing: